via/views/route_by_content/_html_rewriter.py

```diff
@@ -0,0 +1,75 @@
+"""Logic for interacting with HTML rewriters."""
+
+from urllib.parse import parse_qsl, urlencode, urlparse
+
+from pyramid.settings import asbool
+from webob.multidict import MultiDict
+
+from via.configuration import Configuration
```

Comment:
I think something's up with `configuration.py`, which `_html_rewriter.py`'s unit tests don't patch. It's also used by `view_pdf.py` and `view_pdf.py`'s unit tests _do_ patch _except_ for one of its unit tests which depends on the real version. What is `Configuration`? Is it a close collaborator of the pdf and route_by_content views?

Its job seems to be to extract things from `via.*` query params so I think it either belongs in views as a collaborator or it should actually be a validation schema and separately unit-tested (it looks like the kind of thing that'd be a Marshmallow schema in LMS)

Reply:
I wouldn't sweat it right now, as that's candidate #1 to be moved out to a separate library. Given that and the side-effectless behavior, it's mostly a patch when convenient kind of thing.